antarest/study/storage/rawstudy/model/filesystem/config/files.py

```diff
@@ -65,6 +67,78 @@ class FileType(Enum):
     MULTI_INI = "multi_ini"
 
 
+def extract_lines_from_archive(root: Path, posix_path: str) -> t.List[str]:
+    """
+    Extract text lines from various types of files.
+
+    Args:
+        root: 7zip or ZIP file containing the study.
+        posix_path: Relative path to the file to extract.
+
+    Returns:
+        list of lines
+    """
+    if root.suffix.lower() == ".zip":
+        with zipfile.ZipFile(root) as zf:
+            try:
+                with zf.open(posix_path) as f:
+                    text = f.read().decode("utf-8")
+                    return text.splitlines(keepends=False)
+            except KeyError:
+                # File not found in the ZIP archive
+                return []
+    elif root.suffix.lower() == ".7z":
+        with py7zr.SevenZipFile(root, mode="r") as z:
+            try:
+                data = z.read([posix_path])
+                text = data[posix_path].read().decode("utf-8")
+                return text.splitlines(keepends=False)
+            except KeyError:
+                # File not found in the 7z archive
+                return []
+    else:
+        raise ValueError(f"Unsupported file type: {root}")
+
+
+def extract_data_from_archive(
+    root: Path,
+    posix_path: str,
+    reader: IniReader,
+) -> t.Dict[str, t.Any]:
+    """
+    Extract and process data from various types of files.
+
+     Args:
+          root: 7zip or ZIP file containing the study.
+          posix_path: Relative path to the file to extract.
+          reader: IniReader object to use for processing the file.
+
+    Returns:
+        The content of the file, processed according to its type:
+        - SIMPLE_INI or MULTI_INI: dictionary of keys/values
+    """
+    if root.suffix.lower() == ".zip":
```

Comment:
The problem is that we have a lot of code to be refactored regarding reducing the number of functions ... I guess for part, it is better to leave for future iterations.
Otherwise, we will get stock on refactoring all the code.

Reply:
ok as discussed I added a function `read_file_from_archive`